staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go

```diff
@@ -124,6 +124,12 @@ func IsDNS1123Subdomain(value string) []string {
 	if !dns1123SubdomainRegexp.MatchString(value) {
 		errs = append(errs, RegexError(dns1123SubdomainErrorMsg, dns1123SubdomainFmt, "example.com"))
 	}
+	subs := strings.Split(value, ".")
```

Comment:
I think the same could be accomplished with something like:
```
const dns1035LabelFmt string = "[a-z]([-a-z0-9]{0,61}[a-z0-9])?"
```

though all the concerns about tightening validation on everything using this still apply